app/controllers/matches_controller.rb

```diff
@@ -1,6 +1,9 @@
 class MatchesController < ApplicationController
   def index
-    @matches = Match.where(user: current_user)
+    @offers = []
+    Offer.all.each do |offer|
+      if offer.user == current_user { @offers << offer }
```

Comment:
This will crash. an `if` statement takes a condition and then the body after it for multi line if statements, whenever you use a curly brance `{` in ruby it means a block and an IF statement never uses blocks. A better approach would be: 
``` 
@offers << offer if offer.user == current_user
```

An even better approach would be a query using `where` like:

```
@offers = Offer.where(user: current_user)
```

Reply:
I found that solution to be super clumsy too, I didn't pick the where method in order to get multiple results, I probably misread the documentation.